pkg/clients/iam/iamrole.go

```diff
@@ -3,16 +3,54 @@ package iam
 import (
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/iam"
+
+	"github.com/crossplane/stack-aws/apis/identity/v1beta1"
 )
 
 // RoleClient is the external client used for IAMRole Custom Resource
 type RoleClient interface {
 	GetRoleRequest(*iam.GetRoleInput) iam.GetRoleRequest
 	CreateRoleRequest(*iam.CreateRoleInput) iam.CreateRoleRequest
 	DeleteRoleRequest(*iam.DeleteRoleInput) iam.DeleteRoleRequest
+	UpdateRoleRequest(*iam.UpdateRoleInput) iam.UpdateRoleRequest
 }
 
 // NewRoleClient returns a new client using AWS credentials as JSON encoded data.
 func NewRoleClient(conf *aws.Config) (RoleClient, error) {
 	return iam.New(*conf), nil
 }
+
+// GenerateCreateRoleInput from IAMRoleSpec
+func GenerateCreateRoleInput(name string, p *v1beta1.IAMRoleParameters) *iam.CreateRoleInput {
+	m := &iam.CreateRoleInput{
+		RoleName:                 aws.String(name),
+		AssumeRolePolicyDocument: aws.String(p.AssumeRolePolicyDocument),
+		Description:              aws.String(p.Description),
+	}
+	return m
+}
+
+// GenerateUpdateRoleInput from IAMRoleSpec
+func GenerateUpdateRoleInput(name string, p *v1beta1.IAMRoleParameters) *iam.UpdateRoleInput {
+	m := &iam.UpdateRoleInput{
+		RoleName:    aws.String(name),
+		Description: aws.String(p.Description),
+	}
+	return m
+}
+
+// GenerateDeleteRoleInput from IAMRoleSpec
+func GenerateDeleteRoleInput(name string) *iam.DeleteRoleInput {
+	m := &iam.DeleteRoleInput{
+		RoleName: aws.String(name),
+	}
+	return m
+}
+
+// UpdateRoleExternalStatus updates the external status object, given the observation
```

Comment:
Having `External` in the name here as we are actually updating the name of the kubernetes object rather than the external IAM role on AWS. We typically defer this type of functionality to the controller, so this can likely be removed (see comments in controller).

Reply:
Instead we would probably want something like a `GenerateObservation` function, ref: https://github.com/crossplane/stack-aws/blob/94baed407fc79f3a083c409b42542a00f7f0eb74/pkg/clients/rds/rds.go#L220